### PR TITLE
PXP-9636 feat(python3.9):upgrade to python3.9

### PR DESCRIPTION
**Pull request by george42-ctds**
Jira Ticket: [PXP-9636](https://ctds-planx.atlassian.net/browse/PXP-9636)

### Improvements

* Upgrade to Python 3.9.

### CODE REVIEW

File: docs/how-to/deployment.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ ENABLE_AUDIT_LOGS:
 
 The `PUSH_AUDIT_LOGS_CONFIG` field must also be configured.
 
-## Deploying audit-service 1.0.0
+## Deploying audit-service 1.1.0
```
Comment (paulineribeyre) [inline docs/how-to/deployment.md:14]: no need to change this, the doc describes how to go from `v0.*` to `v1.*`, not specifically to 1.1.0. We can leave it as is or change it to `1.*`

File: pyproject.toml [hunk 2]
```diff
@@ -25,6 +25,8 @@ importlib-metadata = {version = "^1.7.0", python = "<3.8"}
 httpx = "^0.12.1"
 psycopg2-binary = "^2.8.5"
 uvicorn = "^0.11.8"
+# 'soft_unicode' was deprecated after version 2.0.1
+markupsafe = "<=2.0.1"
```
[the remark this answers is not in the document]
Comment (paulineribeyre) [reply pyproject.toml:29]: +1, i don't understand the comment, why do we need `soft_unicode` or `markupsafe`?

**Review by paulineribeyre — COMMENTED** [review]
nice!

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (george42-ctds) [reply pyproject.toml:29]: Oh, right, the comments don't fully describe the issue. There is a runtime error when running the app (or executing `alembic upgrade head`) if the markupsafe version has not been pinned. The error is from an import in the jinja2/utils.py

```
from markupsafe import Markup, escape, soft_unicode
ImportError: cannot import name 'soft_unicode' from 'markupsafe'
```

I believe that this is coming from the gen3config/setup.py which requires "jinja2~=3.0.0". The Config object from gen3config is imported in audit-service/config.py.

The `soft_unicode` was [deprecated in newer versions](https://markupsafe.palletsprojects.com/en/2.1.x/changes/#version-2-1-0) of markupsafe. 

Is it okay to leave the pinning in place and change the comments to provide more context?

[the remark this answers is not in the document]
Comment (UchicagoZchen138) [reply pyproject.toml:29]: cool that makes sense, i think we should just provide more context and then its good.

[the remark this answers is not in the document]
Comment (george42-ctds) [reply pyproject.toml:29]: Great! Maybe more verbose comments would help clarify? Something like this?

```
# markupsafe is a jinja dependency. Jinja is used by gen3config. 
# The current version of jinja imports the ‘soft_unicode’ method from markupsafe.
# soft_unicode has been deprecated from markupsafe versions > 2.0.1.
# Pin to <=2.0.1 for compatibility with jinja. 
```

[the remark this answers is not in the document]
Comment (UchicagoZchen138) [reply pyproject.toml:29]: looks good to me

[the remark this answers is not in the document]
Comment (george42-ctds) [reply pyproject.toml:32]: Great, thank you, those changes have been implemented.

[the remark this answers is not in the document]
Comment (george42-ctds) [reply pyproject.toml:32]: Also documenting that the pinning of cryptography has been removed from the pyproject.toml. The base image for the Docker file is no longer using alpine so cryptography can be updated to a newer version.

File: pyproject.toml [hunk 3]
```diff
@@ -25,12 +25,11 @@ importlib-metadata = {version = "^1.7.0", python = "<3.8"}
...
-# Rust code. Alpine 3.7 does not support newer versions of Rust. Pin
-# to <3.4 to fix Dockerfile build.
-# https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst#34---2021-02-07
-# https://cryptography.io/en/latest/installation.html#alpine
-cryptography = "<3.4"
+# markupsafe is a jinja dependency. Jinja is used by gen3config.
+# The current version of jinja imports the ‘soft_unicode’ method from markupsafe.
+# soft_unicode has been deprecated from markupsafe versions > 2.0.1.
+# Pin to <=2.0.1 for compatibility with jinja.
+markupsafe = "<=2.0.1"
```
Comment (paulineribeyre) [inline pyproject.toml:32]: we'll want to update the dependency in gen3config eventually but for now i think this is fine

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**